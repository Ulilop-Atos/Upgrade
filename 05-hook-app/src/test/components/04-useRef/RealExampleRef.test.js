import { shallow } from "enzyme";
import { RealExampleRef } from "../../../components/04-useRef/RealExampleRef";

describe('Tests for <RealExampleRef />', () => { 
 
  const wrapper =  shallow(<RealExampleRef />);

  test('should display properly', () => { 
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('MultipleCustomHooks').exists()).toBe(false);
  });
  
  test('should display MultipleCustomHooks', () => {
    wrapper.find('button').simulate('click'); 
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('MultipleCustomHooks').exists()).toBe(true);    
  });

})