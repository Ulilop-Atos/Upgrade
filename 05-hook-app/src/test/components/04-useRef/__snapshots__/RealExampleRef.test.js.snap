// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for <RealExampleRef /> should display MultipleCustomHooks 1`] = `
<div>
  <h1>
    RealExampleRef
  </h1>
  <hr />
  <MultipleCustomHooks />
  <button
    className="btn btn-primary mt-5"
    onClick={[Function]}
  >
    Show/Hide
  </button>
</div>
`;

exports[`Tests for <RealExampleRef /> should display properly 1`] = `
<div>
  <h1>
    RealExampleRef
  </h1>
  <hr />
  <button
    className="btn btn-primary mt-5"
    onClick={[Function]}
  >
    Show/Hide
  </button>
</div>
`;
